pkg/crc/constants/constants.go

```diff
@@ -111,6 +112,19 @@ func GetDefaultBundle(preset crcpreset.Preset) string {
 	return bundles[runtime.GOOS]
 }
 
+func GetBundleNameFromBundlePath(preset crcpreset.Preset, bundlePath string) string {
+	switch {
+	case strings.HasPrefix(bundlePath, "http://"), strings.HasPrefix(bundlePath, "https://"):
+		return filepath.Base(bundlePath)
+	case strings.HasPrefix(bundlePath, "docker://"):
+		tag := strings.Split(bundlePath, ":")[2]
+		bundles := defaultBundleForOs(preset, tag)
```

Comment:
This does not seem very flexible, this means I can't use `docker://quay.io/crcont/podman-bundle` (it might even crash?), or `docker://quay.io/crcont/openshift-bundle:latest-4.10`. https://github.com/code-ready/snc/pull/569 or https://github.com/code-ready/snc/issues/567 might be better than hardcoding this.
If for now we want to be very strict on the image tag and don't want to add labels, we can also say that the registry name must match the ones on `crcont` and do a registry name -> bundle name conversion rather than using `preset`

Reply:
> This does not seem very flexible, this means I can't use docker://quay.io/crcont/podman-bundle (it might even crash?)

This we can put part of validation when user provide an image without tag, we error out and tell user to use a tag for image. Will add that.

> or docker://quay.io/crcont/openshift-bundle:latest-4.10. 
This doesn't crash but then bundle extraction fail but this can happen with http location where user change the bundle name. Our first goal is to provide an opinionated solution where we tell user what not to use (initially with doc and then with automation). 

> If for now we want to be very strict on the image tag and don't want to add labels, we can also say that the registry name must match the ones on crcont and do a registry name -> bundle name conversion rather than using preset

@cfergeau yes, at this point we are to strict with image tag. I think initially we should go with that and once https://github.com/code-ready/snc/pull/569 merge then we can revisit again.